internal/moreos/moreos_func-e_test.go

```diff
@@ -45,68 +46,107 @@ var (
 	moreosSrcDir embed.FS
 )
 
-// TestProcessGroupAttr_Kill tests sending SIGKILL to fake func-e.
-// On linux, we propagate SIGKILL to the child process as the configured SysProcAttr.Pdeathsig
-// in proc_linux.go.
-func TestProcessGroupAttr_Kill(t *testing.T) {
-	// This works only for linux, sending kill -9 on darwin will not kill the process, we need to kill
-	// via pgid or kill the child first.
-	if runtime.GOOS != "linux" {
-		t.Skip()
+// Test_CallSignals tests sending signals to fake func-e.
+func Test_CallSignals(t *testing.T) {
+	type testCase struct {
+		name           string
+		signal         func(*os.Process) error
+		skip           bool
+		waitForExiting bool
 	}
-	tempDir := t.TempDir()
 
-	// Build a fake envoy and pass the ENV hint so that fake func-e uses it
-	fakeEnvoy := filepath.Join(tempDir, "envoy"+Exe)
-	fakebinary.RequireFakeEnvoy(t, fakeEnvoy)
-	t.Setenv("ENVOY_PATH", fakeEnvoy)
+	tests := []testCase{
+		{
+			name:           "Interrupt",
+			signal:         Interrupt,
+			waitForExiting: true,
+		},
+		{
+			name:           "SIGTERM",
+			signal:         func(proc *os.Process) error { return proc.Signal(syscall.SIGTERM) },
+			waitForExiting: true,
+			// On Windows, os.Process.Signal is not implemented; it will return an error instead of sending
+			// a signal.
+			skip: runtime.GOOS == OSWindows,
+		},
+		{
+			name: "Kill",
+			// On Linux, we propagate SIGKILL to the child process as the configured SysProcAttr.Pdeathsig
+			// in proc_linux.go.
+			signal:         func(proc *os.Process) error { return proc.Kill() },
+			waitForExiting: false, // since the process is killed, it is immediately exit.
+			// This works only for Linux, sending kill -9 on Darwin will not kill the process, we need to
+			// kill via pgid or kill the child first.
+			skip: runtime.GOOS != OSLinux,
+		},
+	}
 
-	fakeFuncE := filepath.Join(tempDir, "func-e"+Exe)
-	requireFakeFuncE(t, fakeFuncE)
+	for _, tc := range tests {
+		tc := tc // pin! see https://github.com/kyoh86/scopelint for why.
 
-	stderr := new(bytes.Buffer)
-	stdout := new(bytes.Buffer)
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.skip {
+				t.Skip()
+			}
 
-	// With an arg so fakeFuncE runs fakeEnvoy as its child and doesn't exit.
-	arg := string(version.LastKnownEnvoy)
-	cmd := exec.Command(fakeFuncE, "run", arg, "-c")
-	cmd.Stdout = stdout
-	cmd.Stderr = stderr
+			tempDir := t.TempDir()
 
-	require.NoError(t, cmd.Start())
+			// Build a fake envoy and pass the ENV hint so that fake func-e uses it
```

Comment:
the more explicit the better 👯 
```suggestion
			// Build a fake envoy and pass it via ENVOY_PATH so that fake func-e uses it
```

Reply:
Sure thing. Updated.